tests/test_salcommander_minimal_controllers.py

```diff
@@ -0,0 +1,145 @@
+# This file is part of ts_SalMultiLanguageTests.
+#
+# Developed for the Rubin Observatory Telescope and Site System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import subprocess
+import pathlib
+import unittest
+import warnings
+
+from lsst.ts import salobj
+
+# Long enough to perform any reasonable operation
+# including starting a CSC or loading a script (seconds)
+STD_TIMEOUT = 60
+
+INITIAL_LOG_LEVEL = 20
+
+index_gen = salobj.index_generator()
+
+languages = ["cpp", "java"]
+
+
+class SalCommanderToMinimalControllerTestCase(unittest.TestCase):
+    """Test a C++ or Java Commander communicating to
+    various minimal controllers."""
+
+    def setUp(self) -> None:
+        salobj.set_random_lsst_dds_partition_prefix()
+        self.index = next(index_gen)
+
+    def getLanguageVariables(self, language) -> None:
+        """Get the language specific test values;
+        either cpp (C++) or java (Java).
+
+        Parameters
+        ----------
+        language : `str`
+            Options are "java" or "cpp" and are case insensitive.
+
+        """
+        if language.lower() == "cpp":
+            commander = "minimal_cpp_commander.sh"
+            language_upper = "C++"
+            initial_log = "initial_log_level="
+        elif language.lower() == "java":
+            commander = "minimal_java_commander.sh"
+            language_upper = "Java"
+            initial_log = "initial logLevel.level "
+        else:
+            raise ValueError(f"Argument {language} must be one of 'cpp' or 'java'")
+        return commander, language_upper, initial_log
```

Comment:
Please consider an enumeration instead of strings. This makes spelling errors less likely.

E.g.:
```
import enum

class Language(enum.Enum):
    CPP = enum.auto()
    JAVA = enum.auto()
```
Also you might consider using a dict, though if/else certainly works. You get an automatic raise if there's no match.
```
def getLanguageVariables(self, language: Language) -> None:
    return {
        Language.CPP: ("minimal_cpp_commander.sh", "initial_log_level="),
        Language.JAVA: ("minimial_java_commander.sh", "initial logLevel.level ")
    }[language]
```
and for language_upper just use language.name

Reply:
Fixed.